server/hotrod/src/main/resources/META-INF/services/org.infinispan.factories.components.ModuleMetadataFileFinder

```diff
@@ -0,0 +1 @@
+org.infinispan.server.hotrod.ServerMetadataFileFinder
```

Comment:
I don't see why passing an empty set is a nice way to signal 'I want all segments'. In my mind empty set means 'I do not want any segments', even if that is a no-op. What's wrong with null or Optional?